multiqc/modules/somalier/somalier.py

```diff
@@ -0,0 +1,616 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from somalier """
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+import csv
+
+import random
+from collections import defaultdict
+from math import isnan, isinf
+from multiqc import config
+from multiqc.plots import bargraph
+from multiqc.plots import heatmap
+from multiqc.plots import scatter
+from multiqc.utils import mqc_colour
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """
+    somalier module class, parses stderr logs.
+    """
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='somalier', anchor='somalier',
+        href='https://github.com/brentp/somalier',
+        info="calculates genotype :: pedigree correspondence checks from sketches derived from BAM/CRAM or VCF")
+
+        # Find and load any somalier reports
+        self.somalier_data = dict()
+        self.somalier_background_pcs = dict()
+        self.somalier_ancestry_cats = list()
+        self.somalier_length_counts = dict()
+        self.somalier_length_exp = dict()
+        self.somalier_length_obsexp = dict()
+
+        # parse somalier sample file
+        for f in self.find_log_files('somalier/samples'):
+            parsed_data = self.parse_somalier_samples(f)
+            if parsed_data is not None:
+                for s_name in parsed_data:
+                    s_name = self.clean_s_name(s_name, f['root'])
+                    if s_name in self.somalier_data.keys():
+                            log.debug("Duplicate sample name found! Overwriting: {}".format(s_name))
+                    self.add_data_source(f, s_name)
+                    self.somalier_data[s_name] = parsed_data[s_name]
+        # parse somalier CSV files # TODO: CSV or TSV?
+        for f in self.find_log_files('somalier/pairs'):
+            parsed_data = self.parse_somalier_pairs_tsv(f)
+            if parsed_data is not None:
+                for s_name in parsed_data:
+                    s_name = self.clean_s_name(s_name, f['root'])
+                    if s_name in self.somalier_data.keys():
+                        log.debug("Duplicate sample name found! Overwriting: {}".format(s_name))
+                    self.add_data_source(f, s_name)
+                    self.somalier_data[s_name] = parsed_data[s_name]
+
+        # parse somalier ancestry files
+        for f in self.find_log_files('somalier/ancestry', filehandles=True):
```

Comment:
the output of somalier is now `${output_prefix} + "somalier-ancestry.tsv"

Reply:
Yup - this is just the key for the search patterns. You can see the full search pattern below:

https://github.com/ewels/MultiQC/pull/1155/files#diff-7f1a8e845586e3f80f703c3a85e4b869R385

```yaml
somalier/somalier-ancestry:
    fn: '*.somalier-ancestry.tsv'
```

(though note the mismatch in the keys)